extensions/src/main/java/org/pentaho/platform/plugin/action/ActionInvokerSystemListener.java

```diff
@@ -48,73 +49,96 @@
   private static final String WORK_ITEM_FILE_EXTENSION = ".json";
   private static final String DEFAULT_CONTENT_FOLDER = "system/default-content";
   private String environmentVariablesFolder;
-
+  private String ERROR = "error";
+  private String OK = "ok";
+  private String WI_STATUS = "/wi-status.";
   //for unit testability
   String getSolutionPath( ) {
     return PentahoSystem.getApplicationContext().getSolutionPath( DEFAULT_CONTENT_FOLDER );
   }
 
+  //places a failure breadcrumb in the specified folder
+  private void placeBreadcrumbFile( String folderPath, boolean success ) {
+    String ext = ERROR;
+    if ( success ) {
+      ext = OK;
+    }
+    String breadcrumbFilename = folderPath + WI_STATUS + ext;
+    File breadcrumb = new File( breadcrumbFilename );
```

Comment:
use File object to construct the breadcrumb file: new File( folderPath, WI_STATUS + etx);